src/DataSet/DataSet.php

```diff
@@ -111,11 +142,86 @@ public function buildIndices() {
 					}
 					$retries++;
 					usleep(100000);
-				} while ($response['indices'][$index]['docs']['num_docs'] != $count);
+				} while ($response['indices'][$index]['docs']['num_docs'] != $documents[$index]);
 			}
 		}
 
 		return $this;
 	}
 
+	/**
+	 * Add mappings to the index
+	 *
+	 * @param string $index
+	 * @return void
+	 */
+	protected function defineMappings($index) {
+		foreach ($this->mappings[$index] as $type => $mappings) {
+			$params = [
+				'index' => $index,
+				'type' => $type
+			];
+
+			if ($this->connection->getConnection()->indices()->existsType($params)) {
+				$this->connection->getConnection()->indices()->deleteMapping($params);
+			}
+			if (empty($mappings)) {
```

Comment:
@dmatsingerzumba You can move this to the beginning of the foreach. To avoid making the call to `existsType`